Makefile

```diff
@@ -250,22 +266,37 @@ depend.update.full: depend.cleanlock depend.update
 depend.cleanlock:
 	-rm Gopkg.lock
 
-depend.update:
+depend.update: | $(DEP)
```

Comment:
what does this pipe thing do?

Reply:
The items that come after the pipe are order-only dependencies.  It basically tells make to care that the file already exists but not to care about how old/new the file is.  Within the last hour or so I made a more complete explanation in another part of the PR about "${ISTIO_BIN}/dep", but in short it's order-only since people might be using a "dep" from another part of the filesystem (outside of ISTIO_BIN) and we just want "make" to care that the file exists (if it was a normal dependency then make would complain that it doesn't have a recipe for the dep that doesn't live in ISTIO_BIN.